test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -1808,7 +1808,7 @@ TEST_P(BasicTest, BackendsRestart) {
   WaitForAllBackends();
   // Stop backends.  RPCs should fail.
   ShutdownAllBackends();
-  CheckRpcSendFailure();
+  CheckRpcSendFailure(num_backends_);
```

Comment:
Please add a comment here indicating that we are sending multiple failed requests instead of just one to ensure that the client notices that all backends are down before we restart them.  If we didn't do this, then a single RPC could fail here due to the race condition between the LB pick and the GOAWAY from the chosen backend being shut down, which would not actually prove that the client noticed that all of the backends are down.  Then, when we send another request below (which we expect to succeed), if the callbacks happen in the wrong order, the same race condition could happen again due to the client not yet having noticed that the backends were all down.